Tickr/Tickr/StockManager.swift

```diff
@@ -0,0 +1,238 @@
+*/
+class StockManager {
+    var isMakingRequest = false
+    var shouldCancelUpdate = false
+    var session = NSURLSession.self()
+    var stockArray = [Stock]()
+    lazy var notificationCenter: NSNotificationCenter = {
+        return NSNotificationCenter.defaultCenter()
+    }()
+    
+    //singleton initialization
+    class var sharedInstance : StockManager {
+        struct Static {
+            static let instance = StockManager()
+        }
+        return Static.instance
+    }
+    
+    
+    /*
+        This will take a search term and try to look up stock symbols or company names based on that term
+        - parameter term: A string to use as a search query
+     */
+    class func fetchStocksFromSearchTerm(term term: String, completion:(stockInfoArray: [StockSearchResult]) -> ()) {
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)) {
+            
+            //sanitize string
+            let query = term.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.URLFragmentAllowedCharacterSet())
+            let url = NSURL(string: "\(Constants.searchURL)\(query!)")
+            
+            //Create NSURLRequest and NSURLSession
+            let request = NSURLRequest(URL: url!)
+            let sessionConfig = NSURLSessionConfiguration.defaultSessionConfiguration()
+            let session = NSURLSession(configuration: sessionConfig)
+            
+            //setup task with completion handler
+            let task : NSURLSessionDataTask = session.dataTaskWithRequest(request, completionHandler: {data, response, error -> Void in
+                
+                //Make sure there was no error
+                guard error == nil else {
+                    print(error!.localizedDescription)
+                    return
+                }
+                
+                do {
+                    let json = try NSJSONSerialization.JSONObjectWithData(data!,
+                        options:NSJSONReadingOptions.MutableContainers) as! NSArray
+                    var stockInfoArray = [StockSearchResult]()
+                    for result in json {
+                        let company = result["company"] as! String
+                        let symbol = result["symbol"] as! String
+                        let newResult = StockSearchResult(symbol: symbol, name: company)
+                        stockInfoArray.append(newResult)
+                    }
+                   completion(stockInfoArray: stockInfoArray)
+                } catch {
+                    print("Error: \(error)")
+                }
+            })
+            
+            //launch session task
+            task.resume()
+        }
+    }
+
+    
+    /*
+        Will fetch current prices from API, and send
+        them to any listeners via NSNotification
+        
+        - parameter stocks: An array of Stock objects
+     */
+    func fetchListOfSymbols(stocks: [Stock]) {
+        if !isMakingRequest {
```

Comment:
this should be removed, imagine you were using this class, and it randomly did not make subsequent requests, plus its not the responsibility of this method to know whether or not is should be executed, its outside the scope of the method